mermithid/misc/FakeTritiumDataFunctions.py

```diff
@@ -293,23 +310,53 @@ def convolved_spectral_rate_arrays(K, Q, mnu, Kmin,
     elif lineshape=='detailed_scattering' or lineshape=='detailed':
+        if resolution_function == 'simulated_resolution' or resolution_function == 'simulated':
+            lineshape_rates = []
+            scale_factors = [ls_params[0]*f for f in ins_res_width_factors]
+            for i in range(len(scale_factors)):
+                lineshape_rates.append(np.flipud(complexLineShape.make_spectrum_simulated_resolution_scaled_fit_scatter_peak_ratio(scale_factors[i], ls_params[1], scatter_peak_ratio_p*p_factors[i], scatter_peak_ratio_q*q_factors[i], scatter_fraction, emitted_peak='dirac')))
+        elif resolution_function == 'gaussian_resolution' or resolution_function == 'gaussian':
+            logger.warn("Scatter peak ratio function for lineshape with Gaussian resolution may not be up-to-date!")
+            gaussian_widths = [ls_params[0]*f for f in ins_res_width_factors]
+            lineshape_rates = [np.flipud(complexLineShape.make_spectrum_gaussian_resolution_fit_scatter_peak_ratio(gaussian_widths[i], ls_params[1], scatter_peak_ratio_p*p_factors[i], scatter_peak_ratio_q*q_factors[i], scatter_fraction, emitted_peak='dirac')) for i in range(len(gaussian_widths))]
+        else:
+            logger.warn('{} is not a resolution function that has been implemented in the FakeDataGenerator'.format(resolution_function))
 
-        lineshape_rates = complexLineShape.spectrum_func_1(K_lineshape/1000., ls_params[0], 0, 1, ls_params[1])
-
-    beta_rates = spectral_rate(K, Q, mnu, final_state_array) #np.zeros(len(K))
-    #for i,ke in enumerate(K):
-    #    beta_rates[i] = spectral_rate(ke, Q, mnu, final_state_array)
+    below_Kmin = np.where(K < Kmin)
 
     #Convolving
-    convolved = convolve(beta_rates, lineshape_rates, mode='same')
-    below_Kmin = np.where(K < Kmin)
-    np.put(convolved, below_Kmin, np.zeros(len(below_Kmin)))
+    if (lineshape=='detailed_scattering' or lineshape=='detailed'):# and (resolution_function == 'simulated_resolution' or resolution_function == 'simulated'):
+        convolved_segments = []
+        beta_rates = spectral_rate(K, Q, mnu, final_state_array)
+        plt.figure(figsize=(7,5))
+        for j in range(len(lineshape_rates)):
+            plt.plot(lineshape_rates[j])
+            #beta_rates = spectral_rate(K_segments[j], Q, mnu, final_state_array)
+            plt.plot(lineshape_rates[j])
+            convolved_j = convolve(beta_rates, lineshape_rates[j], mode='same')
+            np.put(convolved_j, below_Kmin, np.zeros(len(below_Kmin)))
+            #Only including the part of convolved_j that corresponds to the right values of K
+            convolved_segments.append(convolved_j[np.logical_and(Kbounds[j]<=K, K<=Kbounds[j+1])])
+            #convolved.append(convolved_j)
+        convolved = np.concatenate(convolved_segments, axis=None)
+        plt.savefig('varied_lineshapes.png', dpi=200)
```

Comment:
since we only use this function during data generation (so once per experiment) I think leaving the plot in here is ok.